worker/peergrouper/worker.go

```diff
@@ -608,19 +479,66 @@ func (w *pgWorker) updateReplicaSet() (map[string]replicaset.Member, error) {
 		logger.Infof("successfully updated replica set")
 	}
 	if err := setHasVote(removed, false); err != nil {
-		return nil, errors.Annotate(err, "cannot set HasVote removed")
+		return nil, errors.Annotate(err, "removing non-voters")
+	}
+
+	// Return the members from the calculated peer-group that were in the
+	// original known replica-set members.
+	for i := range members {
```

Comment:
I think we want to advertise the "machines that now have voting" and not "machines that used to have voting". And probably the test case is wrong, but we need to add tests for both of those cases.